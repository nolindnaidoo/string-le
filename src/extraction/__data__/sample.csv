name,desc,count
Alpha,First item,1
Beta,Second item,2
Gamma,Third item,3
