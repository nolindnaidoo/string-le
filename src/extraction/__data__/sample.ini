[core]
title = My App Configuration
version = 2.1.3
port = 8080
debug = true
timeout = 30.5
ssl_enabled = false
max_connections = 1000
pi = 3.14159
created_date = 2024-01-15T10:30:00Z
empty_value = 
null_value = null
message = "Welcome to the system!"
description = This is a plain unquoted string
quoted_with_spaces = " padded content "
special_chars = "Contains: semicolons; equals=signs & ampersands"

[database]
host = localhost
port = 5432
name = myapp_production
username = admin_user
password = "secret123!@#"
ssl = true
timeout = 15
max_pool_size = 100
url = postgresql://user:pass@localhost:5432/myapp

[features]
feature_flags = ["analytics", "chat", "notifications"]
deprecated_feature = false
beta_mode = true
rate_limit = 1000
cache_ttl = 3600
maintenance_window = "02:00-04:00 UTC"

[user.alice]
name = Alice Johnson
age = 28
active = true
last_login = 2024-08-17T09:15:22Z
permissions = admin, user, editor
notes = Multi-line content \
  spanning several lines \
  with important details
salary = 85000.50

[user.bob]
name = Bob Smith
age = 35
active = false
department = "Engineering & Development"
skills = "Python, TypeScript, Go"
bio = Simple bio without quotes
temp_access = null
