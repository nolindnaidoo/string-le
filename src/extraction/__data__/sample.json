{
	"greeting": "Hello",
	"nested": { "farewell": "Goodbye" },
	"list": ["One", "Two", 3]
}
