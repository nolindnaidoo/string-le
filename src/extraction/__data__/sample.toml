# TOML Configuration Demo - Comprehensive Test
title = "Enterprise Application Config"
description = "Production configuration with all TOML data types"
version = "2.1.3"

# Numeric values (should be ignored by string extractor)
port = 8080
timeout = 30.5
max_connections = 1000
pi = 3.14159
large_number = 1000000
negative_number = -42

# Boolean values (should be ignored)
debug_mode = true
ssl_enabled = false
maintenance = false

# String values (should be extracted)
app_name = "My Production Application"
environment = "production" 
log_level = "info"
error_message = "Something went wrong during processing"
quoted_with_spaces = " padded content "
unicode_text = "Hello ‰∏ñÁïå! üåç"
special_chars = "Contains: quotes=\"value\" & other stuff"
empty_string = ""

# Date strings (extracted as strings when quoted)
start_time = "08:30:00"
created_date = "2024-08-17T10:30:00Z"

# Arrays of different types
string_array = ["admin", "user", "editor", "guest"]
number_array = [1, 2, 3, 5, 8, 13]
boolean_array = [true, false, true]
mixed_strings = ["text", "more text", "even more"]

# Database configuration
[database]
host = "db-cluster.example.com"
port = 5432
name = "production_database"
username = "app_user"
password = "super!@#$%secret"
ssl_required = true
connection_string = "postgresql://user:pass@localhost:5432/myapp"

# API configuration
[api]
base_url = "https://api.example.com/v2"
version = "v2.1"
timeout = 5000

[api.auth]
type = "oauth2"
client_id = "oauth_12345"
client_secret = "oauth_secret_abcdef"
token_url = "https://auth.example.com/token"

[api.endpoints]
users = "/users"
orders = "/orders/{id}"
payments = "/payments"

# Deep nesting demonstration
[deep.level1.level2.level3]
deeply_nested_string = "Found at maximum depth"
config_value = "Deep configuration setting"
secret_key = "very!@#secret$%^key&*("

# Edge cases that demonstrate robustness
[edge_cases]
colon_in_value = "Contains: colons and semicolons;"
dash_in_value = "Contains - dashes - everywhere"  
bracket_in_value = "Contains [brackets] and {braces}"
special_symbols = "Special: !@#$%^&*()_+-=[]{}|;:,.<>?"
email = "admin@example.com"
url = "https://example.com/path?param=value"
json_like = '{"key": "value", "number": 42}'
sql_query = "SELECT * FROM users WHERE active = true"

# Numeric strings (should be extracted because they're quoted)
string_number = "12345"
string_float = "3.14159"
string_boolean = "true"
string_null = "null"
