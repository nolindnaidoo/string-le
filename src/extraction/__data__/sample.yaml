# YAML Configuration Demo - Comprehensive Test
# This demonstrates all YAML features and edge cases without multiline complexity

# Application metadata
app:
  name: &app_name "Enterprise Application"
  version: "2.1.3"
  description: "Production configuration with all YAML data types"
  environment: "production"

# Numeric values (should be ignored by string extractor)
config:
  port: 8080
  timeout: 30.5
  max_connections: 1000
  pi: 3.14159
  large_number: 1000000
  negative: -42
  
# Boolean values (should be ignored)
flags:
  debug_mode: true
  ssl_enabled: false
  maintenance: false
  beta_features: true
  yes_value: yes
  no_value: no
  on_value: on
  off_value: off

# Null and empty values
nulls_and_empties:
  null_explicit: null
  null_tilde: ~
  null_implicit:
  empty_string: ""
  quoted_empty: ''
  spaces_only: "   "

# String values with various quote styles
strings:
  basic: "Basic quoted string"
  single_quoted: 'Single quoted string'
  unquoted: This is an unquoted string
  with_escapes: "Contains \"quotes\" and \\backslashes\\"
  unicode: "Hello ‰∏ñÁïå! –ó–¥—Ä–∞–≤—Å—Ç–≤—É–π –º–∏—Ä! üåç"
  special_chars: "Special: !@#$%^&*()_+-=[]{}|;:,.<>?"
  json_like: '{"key": "value", "number": 42}'
  url: "https://api.example.com/v2/users"
  email: "admin@example.com"
  multiword: Multiple words without quotes
  padded: " content with padding "

# String representations of other types (should be extracted)
string_representations:
  string_number: "12345"
  string_float: "3.14159"
  string_boolean: "true"
  string_false: "false"
  string_null: "null"
  string_date: "2024-01-15"
  string_zero: "0"
  quoted_number: "42"

# Arrays and sequences
arrays:
  simple_strings: ["admin", "user", "editor", "guest"]
  numbers: [1, 2, 3, 5, 8, 13, 21]
  booleans: [true, false, true, false]
  mixed_strings: ["text", "more text", "even more"]
  flow_style: [one, two, three]
  
# Block sequences
block_sequence:
  - First item
  - Second item with "quotes"
  - Third item with colon
  - Fourth item
  - name: Item with mapping
    value: "Some value"

# Anchors and references
base_config: &base
  host: "localhost"
  port: 5432
  ssl: true
  
database:
  primary:
    <<: *base
    name: "production_database"
    username: "app_user"
    password: "super!@#secret"
    
  replica:
    <<: *base
    name: "replica_database"
    username: "readonly_user"
    password: "replica_secret"
    port: 5433

# Merge keys with overrides
common_settings: &common
  environment: *app_name
  debug: false
  version: "2.1.3"

environments:
  production:
    <<: *common
    log_level: "warning"
    monitoring: "https://monitoring.example.com"
    
  staging:
    <<: *common
    log_level: "info"
    test_mode: true
    
  development:
    <<: *common
    debug: true
    log_level: "debug"

# Complex nested structures
api:
  base_url: "https://api.example.com"
  version: "v2.1"
  auth:
    type: "oauth2"
    client_id: "oauth_12345"
    client_secret: "oauth_secret_abcdef"
    scopes: ["read", "write", "admin"]
  endpoints:
    users: "/users"
    orders: "/orders/{id}"
    payments: "/payments"
  timeouts:
    connect: 5000
    read: 30000
    total: 60000

# Edge cases and problematic content
edge_cases:
  colon_in_value: "Contains: colons and semicolons;"
  dash_in_value: "Contains - dashes - everywhere"  
  bracket_in_value: "Contains [brackets] and {braces}"
  pipe_in_value: "Contains | pipes | for separation"
  ampersand_in_value: "Contains & ampersands & references"
  percent_in_value: "Contains % percent % signs"
  at_symbol: "@username and @mentions"
  hash_symbol: "#hashtags and #comments"
  
# Complex real-world examples
user_permissions:
  admin: &admin_perms
    - "users:read"
    - "users:write"
    - "users:delete"
    - "orders:*"
    - "system:admin"
    
  editor:
    - "users:read"
    - "users:write"
    - "orders:read"
    - "orders:write"
    
  viewer:
    - "users:read"
    - "orders:read"

# Microservices configuration
microservices:
  auth:
    image: "auth-service:v2.1.3"
    replicas: 3
    config:
      jwt_secret: "jwt_super_secret_key_12345"
      token_expiry: "24h"
      refresh_expiry: "7d"
      
  api_gateway:
    image: "api-gateway:v1.5.2"
    replicas: 2
    config:
      rate_limit: "1000/hour"
      cors_origins: ["https://app.example.com", "https://admin.example.com"]
      
  notification:
    image: "notification-service:v1.2.1"
    replicas: 1
    config:
      smtp_host: "smtp.example.com"
      smtp_user: "notifications@example.com"
      webhook_url: "https://hooks.example.com/notify"
