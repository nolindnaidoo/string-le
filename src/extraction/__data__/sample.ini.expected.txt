["analytics", "chat", "notifications"]
02:00-04:00 UTC
100
1000
15
2.1.3
2024-01-15T10:30:00Z
2024-08-17T09:15:22Z
28
3.14159
30.5
35
3600
5432
8080
85000.50
admin_user
admin, user, editor
Alice Johnson
Bob Smith
Contains: semicolons; equals=signs & ampersands
Engineering & Development
localhost
Multi-line content \
My App Configuration
myapp_production
padded content
postgresql://user:pass@localhost:5432/myapp
Python, TypeScript, Go
secret123!@#
Simple bio without quotes
This is a plain unquoted string
Welcome to the system!
